### PR TITLE
Feature/split game instance in sub systems

### PR DESCRIPTION
**Pull request by TheEmidee**

### CODE REVIEW

File: Source/GameBaseFramework/Private/Engine/SubSystems/GBFGameInstanceGameStateSystem.cpp [hunk 1]
```diff
@@ -82,6 +82,15 @@ bool UGBFGameInstanceGameStateSystem::IsStateWelcomeScreenState( const UGBFGameS
     return state != nullptr && Settings->WelcomeScreenGameState.Get() == state;
 }
 
+UGBFGameState * UGBFGameInstanceGameStateSystem::GetGameStateFromName( FName state_name ) const
```
Comment (WannesVanHooste) [inline Source/GameBaseFramework/Private/Engine/SubSystems/GBFGameInstanceGameStateSystem.cpp:85]: Why removal of the const UGBFGameState?

File: Source/GameBaseFramework/Public/Engine/SubSystems/GBFGameInstanceGameStateSystem.h [hunk 2]
```diff
@@ -35,11 +35,10 @@ class GAMEBASEFRAMEWORK_API UGBFGameInstanceGameStateSystem : public UGameInstan
     void Initialize( FSubsystemCollectionBase & collection ) override;
     void UpdateCurrentGameStateFromCurrentWorld();
     bool IsStateWelcomeScreenState( const UGBFGameState * state ) const;
+    UGBFGameState * GetGameStateFromName( FName state_name ) const;
```
Comment (WannesVanHooste) [inline Source/GameBaseFramework/Public/Engine/SubSystems/GBFGameInstanceGameStateSystem.h:38]: why removal of const?

File: Source/GameBaseFramework/Public/Engine/SubSystems/GBFGameInstanceCoreDelegatesSubsystem.h [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    
+    uint8 ItIsLicensed : 1;
+};
+
+FORCEINLINE bool UGBFGameInstanceCoreDelegatesSubsystem::IsLicensed() const
+{
+    return ItIsLicensed;
+}
+
+FORCEINLINE UGBFGameInstanceCoreDelegatesSubsystem::FOnAppReactivatedOrForegroundDelegate & UGBFGameInstanceCoreDelegatesSubsystem::OnAppReactivatedOrForeground()
```
Comment (VectorGamez) [inline Source/GameBaseFramework/Public/Engine/SubSystems/GBFGameInstanceCoreDelegatesSubsystem.h:55]: I think this and the next function can both be const

File: Source/GameBaseFramework/Public/Engine/SubSystems/GBFGameInstanceGameStateSystem.h (hunk 2 above)
Comment (TheEmidee) [reply Source/GameBaseFramework/Public/Engine/SubSystems/GBFGameInstanceGameStateSystem.h:38]: because in some of the calls we need a non const game state

**Pull request merged**